### PR TITLE
Mic's PR

### PR DESCRIPTION
**Pull request by mic-css**

### CODE REVIEW

File: lib/player.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  ERROR_MESSAGE = 'Invalid weapon'
+  attr_reader :name, :weapon
+
+  def initialize(name)
+    @name = name
+    @weapon = nil
+  end
+
+  def choose_weapon(weapon)
+    raise ERROR_MESSAGE unless Game::WEAPONS.include? weapon
```
Comment (mic-css) [inline lib/player.rb]: [Ruby style guide](https://github.com/bbatsov/ruby-style-guide#prefer-raise-over-fail) recommends the opposite.

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  def initialize(player_1, player_2, multiplayer = false)
+    @players     = [player_1, player_2]
+    @multiplayer = multiplayer
+  end
+
+  def player(number)
+    message = 'Please specify a valid player number'
+    raise message if !number.is_a?(Numeric)
+    raise message if (number > players.size || number <= 0 )
```
Comment (mic-css) [inline lib/game.rb]: [Ruby style guide](https://github.com/bbatsov/ruby-style-guide#prefer-raise-over-fail) recommends the opposite.

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  def initialize(player_1, player_2, multiplayer = false)
+    @players     = [player_1, player_2]
+    @multiplayer = multiplayer
+  end
+
+  def player(number)
+    message = 'Please specify a valid player number'
+    raise message if !number.is_a?(Numeric)
```
Comment (mic-css) [inline lib/game.rb]: [Ruby style guide](https://github.com/bbatsov/ruby-style-guide#prefer-raise-over-fail) recommends the opposite.

**Comment by silvabox** [comment]
```
  def choose_weapon(weapon)
    weapon = weapon.to_sym
    # TODO: redundant
    raise ARGUMENT_ERROR_MSG unless Game::WEAPONS.include? weapon
    @weapon = weapon_klass.new(weapon)
  end
```
Should the validation of the weapon type be in the Player or the Weapon?

**Comment by silvabox** [comment]
Great work though - a really thorough solution!

**Pull request closed without merging**